gorelease changelog failes with gitea

The `gorelease` action fails when running against a `gitea` instance.
The .gorelease.yml configuration snippet is minimalistic:
```
changelog:
  use: gitea
gitea_urls:
  api: https://gitea...../api/v1
  download: https://gitea.....
  # set to true if you use a self-signed certificate
  skip_tls_verify: true
```

and the output is following:
```
• generating changelog
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0xe4c302]
goroutine 49 [running]:
github.com/goreleaser/goreleaser/v2/internal/client.(*giteaClient).Changelog(0xa57877?, 0xc000cbf770?, {{0xc0007b3890, 0x9}, {0xc0008676a6, 0x16}, {0x0, 0x0}, {0x0, 0x0}, ...}, ...)
	github.com/goreleaser/goreleaser/v2@v2.5.1/internal/client/gitea.go:89 +0x1a2
github.com/goreleaser/goreleaser/v2/internal/pipe/changelog.(*scmChangeloger).Log(0x5f06a0?, 0xc00028a888)
	github.com/goreleaser/goreleaser/v2@v2.5.1/internal/pipe/changelog/changelog.go:453 +0xa5
github.com/goreleaser/goreleaser/v2/internal/pipe/changelog.buildChangelog(0xc00028a888)
	github.com/goreleaser/goreleaser/v2@v2.5.1/internal/pipe/changelog/changelog.go:[26](https://gitea......./developer/goreleaser-hello-world/actions/runs/8#jobstep-7-26)2 +0x37
github.com/goreleaser/goreleaser/v2/internal/pipe/changelog.Pipe.Run({}, 0xc000[28](https://gitea......./developer/goreleaser-hello-world/actions/runs/8#jobstep-7-28)a888)
	github.com/goreleaser/goreleaser/v2@v2.5.1/internal/pipe/changelog/changelog.go:86 +0x185
github.com/goreleaser/goreleaser/v2/cmd.releaseProject.func1.Handle.1(0xc00011a520?)
	github.com/goreleaser/goreleaser/v2@v2.5.1/internal/middleware/errhandler/error.go:15 +0x1b
github.com/goreleaser/goreleaser/v2/cmd.releaseProject.func1.Log.2(0xc00028a888)
	github.com/goreleaser/goreleaser/v2@v2.5.1/internal/middleware/logging/logging.go:[30](https://gitea......./developer/goreleaser-hello-world/actions/runs/8#jobstep-7-30) +0x132
github.com/goreleaser/goreleaser/v2/internal/middleware/skip.Maybe.func1(0xc00028a888)
	github.com/goreleaser/goreleaser/v2@v2.5.1/internal/middleware/skip/skip.go:42 +0x10c
github.com/goreleaser/goreleaser/v2/cmd.releaseProject.func1()
	github.com/goreleaser/goreleaser/v2@v2.5.1/cmd/release.go:123 +0x182
github.com/caarlos0/ctrlc.(*Ctrlc).Run.func1()
	github.com/caarlos0/ctrlc@v1.2.0/ctrlc.go:48 +0x23
created by github.com/caarlos0/ctrlc.(*Ctrlc).Run in goroutine 1
	github.com/caarlos0/ctrlc@v1.2.0/ctrlc.go:47 +0x94
::error::The process '/opt/hostedtoolcache/goreleaser-action/2.5.1/x64/goreleaser' failed with exit code 2
```

--- a/internal/client/gitea.go
+++ b/internal/client/gitea.go
@@ -83,13 +83,16 @@ func (c *giteaClient) Changelog(_ *context.Context, repo Repo, prev, current str
 	var log []ChangelogItem
 
 	for _, commit := range result.Commits {
-		log = append(log, ChangelogItem{
-			SHA:            commit.SHA,
-			Message:        strings.Split(commit.RepoCommit.Message, "\n")[0],
-			AuthorName:     commit.Author.FullName,
-			AuthorEmail:    commit.Author.Email,
-			AuthorUsername: commit.Author.UserName,
-		})
+		item := ChangelogItem{
+			SHA:     commit.SHA,
+			Message: strings.Split(commit.RepoCommit.Message, "\n")[0],
+		}
+		if author := commit.Author; author != nil {
+			item.AuthorName = author.FullName
+			item.AuthorEmail = author.Email
+			item.AuthorUsername = author.UserName
+		}
+		log = append(log, item)
 	}
 	return log, nil
 }

--- a/internal/client/gitea_test.go
+++ b/internal/client/gitea_test.go
@@ -614,6 +614,14 @@ func TestGiteaChangelog(t *testing.T) {
 				Commits: []*gitea.Commit{
 					{
 						CommitMeta: &gitea.CommitMeta{
+							SHA: "2efbc15d0904f5f966355967a4eedd61a8006660",
+						},
+						RepoCommit: &gitea.RepoCommit{
+							Message: "chore: commit without author",
+						},
+					},
+					{
+						CommitMeta: &gitea.CommitMeta{
 							SHA: "c8488dc825debca26ade35aefca234b142a515c9",
 						},
 						Author: &gitea.User{
@@ -650,6 +658,10 @@ func TestGiteaChangelog(t *testing.T) {
 	result, err := client.Changelog(ctx, repo, "v1.0.0", "v1.1.0")
 	require.NoError(t, err)
 	require.Equal(t, []ChangelogItem{
+		{
+			SHA:     "2efbc15d0904f5f966355967a4eedd61a8006660",
+			Message: "chore: commit without author",
+		},
 		{
 			SHA:            "c8488dc825debca26ade35aefca234b142a515c9",
 			Message:        "feat: impl something",

